docs/pages/server-access/guides/azure-discovery.mdx

```diff
@@ -0,0 +1,271 @@
+---
+title: Configure Teleport to Automatically Enroll Azure Virtual Machines (Preview)
+description: How to configure Teleport to automatically enroll Azure Virtual Machines.
+---
+
+The Teleport Discovery Service can connect to Azure and automatically
+discover and enroll virtual machines matching configured labels. It will then
+execute an install script on these discovered instances that will install
+Teleport, start it and join the cluster.
+
+In this scenario, Teleport Discovery Service uses an Azure invite token with a
+long time-to-live (TTL), so that new instances can be discovered and added to
+the Teleport cluster for the lifetime of the token.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- Azure subscription with virtual machines and permissions to create and attach
+managed identities.
+- Azure virtual machines running Ubuntu/Debian/RHEL if making use of the
+default Teleport install script. (For other Linux distributions, you can
+install Teleport manually.)
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/5. Create an Azure invite token
+
+When discovering Azure Virtual Machines, Teleport makes use of Azure invite tokens for
+authenticating joining Nodes.
+
+Create a file called `token.yaml`:
+
+```yaml
+# token.yaml
+kind: token
+version: v2
+metadata:
+  # the token name is not a secret because instances must prove that they are
+  # running in your Azure subscription to use this token
+  name: azure-discovery-token
+  # set a long expiry time, as the default for tokens is only 30 minutes
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  # use the minimal set of roles required
+  roles: [Node]
+
+  # set the join method allowed for this token
+  join_method: azure
+
+  azure:
+    allow:
+    # specify the Azure subscription which Nodes may join from
+    - subscription: "123456789"
+```
+
+Assign the `subscription` field to your Azure subscription ID.
+Add the token to the Teleport cluster with:
+
+```code
+$ tctl create -f token.yaml
+```
+
+## Step 2/5. Configure IAM permissions for Teleport
+
+The Teleport Discovery Service needs Azure IAM permissions to discover and register Azure virtual machines.
+
+### Configure an Azure service principal
+
+(!docs/pages/includes/server-access/azure-configure-service-principal.mdx!)
```

Comment:
Unless we have plans to use this partial elsewhere, I would suggest inlining it, since we only use it in one place.